src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -777,13 +775,18 @@ public CaseWorkerProfile processExistingCaseWorkers(
             StaffProfileCreationRequest cwUiRequest, CaseWorkerProfile caseWorkerProfiles) {
 
         CaseWorkerProfile filteredProfile = null;
-
+        if (cwUiRequest.getSuspended() == null) {
+            cwUiRequest.setSuspended(caseWorkerProfiles.getSuspended());
+        }
         if (cwUiRequest.isSuspended()) {
             //when existing profile with delete flag is true in request then suspend user
             if (isUserSuspended(UserProfileUpdatedData.builder().idamStatus(IDAM_STATUS_SUSPENDED).build(),
                     caseWorkerProfiles.getCaseWorkerId(), ORIGIN_EXUI)) {
                 caseWorkerProfiles.setSuspended(true);
             }
+        } else if (Boolean.TRUE.equals(caseWorkerProfiles.getSuspended()) && isUserSuspended(UserProfileUpdatedData
```

Comment:
scenario:
In  the request suspended flag = null  , case worker data base suspended flag = true
then line no 779 we are setting cwUiRequest.suspended = true
which will execute line 787 to update the IDAM and UP .  i think this call is not required. correct me if my understanding is wrong.

Reply:
@vasthav-patta the question is unclear as its the same question copied from above !!! kindly refer

Reply:
As per discussion with @sahitya-desireddy @praveen saxena , we agreed these repetitive calls exists when there is no changes in the suspended state. This will be handled in the separate ticket.